src/measurement.py

```diff
@@ -7,26 +18,22 @@ class Measurement:
     and are not drawn.
     """
 
-    def __init__(self, x: float, y: float):
+    def __init__(self, x: float, x_lower_error: float, x_upper_error: float, y: float, y_lower_error: float, y_upper_error: float):
         self.x: float = x
         self.y: float = y
-        self.lower_error: float = 0.0
-        self.upper_error: float = 0.0
-
-    def add_error_bounds(self, lower_error: float, upper_error: float):
-        self.lower_error = lower_error
-        self.upper_error = upper_error
+        self.x_error: Error = Error(x_lower_error, x_upper_error)
+        self.y_error: Error = Error(y_lower_error, y_upper_error)
 
     def data(self) -> tuple[float, float]:
         return (self.x, self.y)
 
     def error_bounds(self) -> tuple[float, float]:
         return (self.lower_error, self.upper_error)
 
-    def has_error_bounds(self) -> bool:
-        return self.lower_error != 0.0 or self.upper_error != 0.0
+    #def has_error_bounds(self) -> bool:
+    #    return self.lower_error != 0.0 or self.upper_error != 0.0
```

Comment:
Is this still needed? If not, get rid of it. If yes, uncomment it.

Reply:
Done